src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -37,6 +37,85 @@ public AnalyzeCommandBaseTests(ITestOutputHelper output)
             Output.WriteLine($"The seed that will be used is: {TestRule.s_seed}");
         }
 
+        [Fact]
+        public void AnalyzeCommand_AbsoluteFileURIWithPathsContainsIllegalCharacters()
+        {
+            var logger = new TestMessageLogger();
+            var command = new TestMultithreadedAnalyzeCommand();
+
+            // No content is set for the following uri, so when its length or content is retrieved,
+            // the file path format will be evaluated.
+            var uri1 = new Uri("C:\\test" + Path.GetInvalidPathChars()[0] + "ABC.md", UriKind.Absolute);
+            var target1 = new EnumeratedArtifact(FileSystem.Instance) { Uri = uri1 };
```

Comment:
>EnumeratedArtifact [](http://example.com/codeflow?start=30&length=18)

this doesn't this code crash? doesn't that make more sense as a design?

i.e., why should we allow a URI with invalid characters to be successfully created? is it so that users can have the convenience of creating illegal file URIs in cases where they bring their own stream? is that really a compelling user scenario?